### PR TITLE
fix hash bug in cmllr flow

### PR DESCRIPTION
**Pull request by JackTemaki**
After discussing with @michelwi and the affected people @Marvin84 and @christophmluscher we came to the conclusion that it is better to not fix this issue "backwards compatible (example fix in https://github.com/rwth-i6/i6_core/tree/nick_cmllr_true_fix)", but just as is.

This will invalidate all existing HMM-GMM SAT training run graphs, which would have to be migrated before updating.

Due to the severity of this fix I would prefer if all reviewers would accept before merging.

### CODE REVIEW

File: sat/flow.py [hunk 1]
```diff
@@ -42,7 +44,13 @@ def add_cmllr_transform(
     multiply = net.add_node(
         "signal-matrix-multiplication-f32",
         "apply-cmllr-transform",
-        {"file": os.path.join(str(transform_dir), matrix_name)},
+        {
+            "file": DelayedFormat(
+                "{transform_dir}/{matrix_name}",
+                transform_dir=transform_dir,
+                matrix_name=matrix_name,
+            )
```
Comment (michelwi) [inline sat/flow.py:52]: if `transform_dir` is a Path, then you should be able to use 
```
transform_dir.join_right(matrix_name)
```
to simplify

Comment (JackTemaki) [reply sat/flow.py:52]: Okay this function is new to me...

Comment (JackTemaki) [reply sat/flow.py]: Okay, changed it and added docstrings (I think we should do this for every code we touch now)

**Review by michelwi — APPROVED** [review]

**Review by Marvin84 — APPROVED** [review]

**Review by christophmluscher — APPROVED** [review]

**Comment by JackTemaki** [comment]
@curufinwe any last objection? If not I will merge this soon.

**Pull request merged**